### PR TITLE
All Xenos Now Have Overwatch Ability Through Chat

### PR DESCRIPTION
**Pull request by KornFlaks**
## About The Pull Request

All main caste xenos (not minions) and larva now have the ability to overwatch other xenos through chat (F) button.

However, only the queen is able to arbitrarily select a xeno from the hive to overwatch without seeing their voice in the chat (simply to prevent increasing the number of buttons).

## Why It's Good For The Game

Now larva and young runners can rest inside the safety of the silo and grab popcorn to watch the rest of the hive get railed at the front line without having to actually risk their own lives to see how it is going.

image

The drone is the player in this image and can now "overwatch" the queen via the (F) button in the chat. However, only the queen's overwatch will result in the eye appearing on the xeno HUD overlay.

Requires #10250 to be merged in order to function for all xenos other than the queen.

## Changelog
:cl:
add: All xenos now have the ability to overwatch other xenos through the chat (F) button.
/:cl:

### CODE REVIEW

**Comment by XSlayer300** [comment]
But wait, how about Hiveminds and Shrikes? Can they still view xenos without them talking?

**Comment by KornFlaks** [comment]
Shrikes never had overwatch. With this PR, now they do but they dont get the list though, only queen does.

Hivemind are completely unchanged. This PR does not change anything related to hiveminds.

**Comment by FosterGrime-Heymynameslouie** [comment]
blessed pr spped merhg

**Comment by Naaanii** [comment]
Hmm, I think this would be better to be given to hive leaders only.

**Review by BraveMole — CHANGES_REQUESTED** [review]
this pr does not what you claim it does, I think you confused both prs.

Anyway, adding a button to everyone sucks a bit. Giving everyone hive watch, yes, but that should be done use hivemenu

File: code/modules/mob/living/carbon/xenomorph/xenoprocs.dm [hunk 1]
```diff
@@ -183,6 +183,20 @@
...
+		if(!check_state())
+			return
+		var/follow_name = href_list["watch_xeno_name"]
+		for(var/mob/living/carbon/xenomorph/X AS in hive.get_watchable_xenos(usr))
+			if(isnum(X.nicknumber))
+				if(num2text(X.nicknumber) != follow_name)
+					continue
+			else
+				if(X.nicknumber != follow_name)
+					continue
```
Comment (BraveMole) [inline code/modules/mob/living/carbon/xenomorph/xenoprocs.dm:196]: you can concatenate that with && and else if

**Review by BraveMole — CHANGES_REQUESTED** [review]

**Comment by Durbles** [comment]
This is a very good change.

**Comment by private-tristan** [comment]
before:
> "tadpole landing" - young runner (305)
"where" 
"where?" 

after:
> "tadpole landing" - young runner (305)
"cool"

**Review by BraveMole — APPROVED** [review]

**Comment by Al-1ce** [comment]
> Shrikes never had overwatch. With this PR, now they do but they dont get the list though, only queen does.
> 
> Hivemind are completely unchanged. This PR does not change anything related to hiveminds. Hivemind never had a list either. Players always had to ask someone to talk in chat in order to jump to them.

I think giving shrike the eye-icon or a variation of it would be fair.

**Pull request merged**